internal/io/chunk_writer.go

```
@@ -0,0 +1,151 @@
+	w := &chunkWriter{
+		l:         l,
+		chunkSize: (1 << 10) * 1024, // default chunk size is 1MB
+		writer:    writer,
+	}
+	for _, opt := range opts {
+		opt(w)
+	}
+	return w
+}
+
+// Write writes data to the writer in chunks
+func (w *chunkWriter) Write(p []byte) (n int, err error) {
+	if w.writerTmp == nil {
+		// create a temporary writer
+		cleanFunc, err := w.initTmpWriter()
+		if err != nil {
+			return 0, err
+		}
+		// register the cleanup function
+		w.cleanTmp = cleanFunc
+	}
+
+	// write the data to the temporary writer
+	n, err = w.writerTmp.Write(p)
+	if err != nil {
+		return 0, err
+	}
+	w.size += n
+
+	// if the size exceeds the chunk size, write to the actual writer
+	if w.size >= w.chunkSize {
+		if err := w.Flush(); err != nil {
+			return n, err
+		}
+	}
+	return n, nil
+}
+
+// Flush flushes the data to the actual writer
+func (w *chunkWriter) Flush() error {
+	// perform flush on the temporary writer
+	// if it implement Flusher
+	if f, ok := w.writerTmp.(WriteFlusher); ok {
+		if err := f.Flush(); err != nil {
+			return err
+		}
+	}
+
+	// get the reader converter based on the extension
+	reader, cleanFunc, err := w.getConvertedReader()
+	if err != nil {
+		return err
+	}
+	defer cleanFunc()
+
+	// write the converted data to the actual writer
+	n, err := io.Copy(w.writer, reader)
+	if err != nil {
+		return err
+	}
+	w.written += n
+
+	// reset
+	w.size = 0
+	w.writerTmp = nil
+	w.cleanTmp()
+
+	return nil
+}
+
+// Close closes the writer and cleans up the temporary file
+func (w *chunkWriter) Close() error {
+	w.cleanTmp()
+	return nil
+}
+
+// initTmpWriter initializes the temporary writer
+func (w *chunkWriter) initTmpWriter() (func(), error) {
+	// create a temporary file with the specified extension
+	f, err := os.CreateTemp(os.TempDir(), fmt.Sprintf("*.%s", w.extension))
+	if err != nil {
+		return nil, err
+	}
+	w.writerTmp = f
+
+	return func() {
+		f.Close()
+		os.Remove(f.Name())
```

we don't use errors returned from these functions?

---

I'm just thinking to suppress all unrelated internal errors that it's ok to suppress. If there's any other scenario that makes this error showing important, we can return the error